warp/yul/ExecEnv.py

```diff
@@ -24,30 +24,26 @@ def __init__(self):
         self.needs_exec_env: list[str] = []
         self.visited: list[str] = []
 
-    def traverse_inner(self, calledFuncs):
-        if calledFuncs == []:
-            return
-        else:
-            for func in calledFuncs:
+    def traverse_callgraph(self, callgraph):
+        for k, v in callgraph.items():
+            stack = []
+            stack.append(k)
+
+            while len(stack):
+                func = stack[-1]
+                stack.pop()
+
                 if func in self.visited:
                     continue
-                else:
-                    if self.callgraph[func]["calldataOp"]:
-                        self.needs_exec_env.append(func)
-                    self.visited.append(func)
-                    self.traverse_inner(self.callgraph[func]["calledFunctions"])
 
-    def traverse_callgraph(self, callgraph):
-        for k, v in callgraph.items():
-            if k == "fun_ENTRY_POINT":
-                for func in v["calledFunctions"]:
-                    try:
+                try:
+                    if callgraph[func]["calldataOp"]:
                         self.needs_exec_env.append(func)
-                        self.traverse_inner(self.callgraph[func]["calledFunctions"])
-                    except KeyError:
-                        continue
-            else:
-                continue
+                    self.visited.append(func)
+                    stack.extend(v["calledFunctions"])
+                except KeyError:
```

Comment:
What does that `KeyError` refer to? `callgraph[func]`, `callgraph[func]["calldataOp"]` or `v["calledFunctions"]`? Would it be more clear to use `dict.get` instead of []?

Reply:
The `KeyError` refers to `callgraph[func]` so it seems easier to use `KeyError` unless you know something equivalent to [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) from JavaScript for Python

Reply:
Unfortunately, there isn't builtin optional chaining, AFAIK. 

There are some options though:
1. `callgraph.get(func, {}).get("calldataOp")` or 
2. Assign and check: `info = callgraph.get(func); if not info: continue`
3. If we were able to use assignment expressions from python3.8 `if info := callgraph.get(func)`
4. Catch `KeyError` as you did

All these options have a difference balance of terseness and clarity. It's your call)